### PR TITLE
Product search by handle and get first variant of bundle product

### PR DESCRIPTION
**Pull request by OniiCoder**

### CODE REVIEW

File: src/Requests/Products/GetProductByHandle.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        //
+    }
+
+    public function graphQuery(): string
+    {
+        return <<<QUERY
+            productByHandle(handle: "$this->productHandle") {
+                id
+                variants(first: 1) {
```
Comment (tonning) [inline src/Requests/Products/GetProductByHandle.php]: Can we up this to 250 (the max)?

File: src/Resources/Products.php [hunk 2]
```diff
@@ -54,4 +55,23 @@ public function get(string $productId): ProductResponse
...
+    {
+        $response = $this->api->send(new GetProductByHandle($productHandle));
+
+        $this->validate($response);
+
+        if ($response->json('data.productByHandle') === null) {
+            throw new GraphQlException('Product not found');
+        }
+
+        return new ProductResponse($response->json('data.productByHandle'));
```
Comment (tonning) [inline src/Resources/Products.php]: Not sure the `GraphQlException` is the right exception. There's technically nothing wrong with the GraphQL query, it's just that no products where found.

What if it just returns a `ProductResponse` with no product? Not sure what the best approach to this is.

```suggestion
        return new ProductResponse($response->json('data.productByHandle') ?? []);
```

**Review by tonning — CHANGES_REQUESTED** [review]

File: src/Requests/Products/GetProductByHandle.php (hunk 1 above)
Comment (OniiCoder) [reply src/Requests/Products/GetProductByHandle.php]: Okay

**Comment by nrthbound** [comment]
Is this still a draft or is it ready for review?

**Comment by OniiCoder** [comment]
Ready now @nrthbound

**Review by nrthbound — APPROVED** [review]

**Review by tonning — APPROVED** [review]

**Pull request merged**